packages/iTwinUI-react/src/core/Dialog/DialogBackdrop.tsx

```diff
@@ -0,0 +1,209 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import ReactDOM from 'react-dom';
+import cx from 'classnames';
+import {
+  CommonProps,
+  useTheme,
+  getContainer,
+  getDocument,
+  FocusTrap,
+} from '../utils';
+import '@itwin/itwinui-css/css/dialog.css';
+import { CSSTransition } from 'react-transition-group';
+
+export type DialogBackdropProps = {
+  /**
+   * Flag whether dialog should be shown.
+   * @default false
+   */
+  isOpen?: boolean;
+  /**
+   * Handler that is called when Dialog is closed.
+   */
+  onClose?: (event?: React.SyntheticEvent) => void;
+  /**
+   * Flag whether dialog is dismissible. If false, you can't close it.
+   * @default true
+   */
+  isDismissible?: boolean;
+  /**
+   * Flag whether dialog should be closed on background overlay press.
+   * @default true
+   */
+  closeOnExternalClick?: boolean;
+  /**
+   * Flag whether dialog should be closed on Escape key press.
+   * @default true
+   */
+  closeOnEsc?: boolean;
+  /**
+   * Handle key press. Returns the keyboard event.
+   */
+  onKeyDown?: React.KeyboardEventHandler;
+  /**
+   * Id of the root where the dialog will be rendered in.
+   * @default 'iui-react-portal-container'
+   */
+  dialogRootId?: string;
+  /**
+   * Document where the dialog will be rendered.
+   * Can be specified to render in a different document (e.g. a popup window).
+   * @default document
+   */
+  ownerDocument?: Document;
+  /**
+   * Type of the dialog.
+   * @default 'default'
+   */
+  styleType?: 'default' | 'fullPage';
```

Comment:
I know it's not exposed yet but we might want to rename this prop. I'm not sure why this prop exists on backdrop (should be on `Dialog` only) but it will cause confusion when we add draggable dialog (viewport and relative cases).

Reply:
The problem is that everything is done trough backdrop in CSS.